File: app/src/main/java/com/kakacat/minitool/currencyconversion/CountryFragment.java
package com.kakacat.minitool.currencyconversion;

import android.content.Context;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.facebook.drawee.view.SimpleDraweeView;
import com.kakacat.minitool.R;
import com.kakacat.minitool.common.ui.view.MyPopupWindow;
import com.kakacat.minitool.currencyconversion.model.CountryBean;

import java.util.List;

public class CountryFragment extends Fragment {

    private Contract.View mainView;
    private List<CountryBean> countryBeanList;

    public EditText et;
    public CountryBean countryBean;
    private TextView tvCountryName;
    private TextView tvMoneyUnit;
    private SimpleDraweeView sdv;
    private MyPopupWindow popupWindow;
    private Context context;
    private int flag;

    public CountryFragment(Contract.View mainView, List<CountryBean> countryBeanList, int flag) {
        this.mainView = mainView;
        this.countryBeanList = countryBeanList;
        this.flag = flag;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        if (this.context == null) {
            this.context = getContext();
        }
        View view = inflater.inflate(R.layout.country_view_layout, container, false);

        if (flag == 1) {
            countryBean = countryBeanList.get(0);
        } else if (flag == 2) {
            countryBean = countryBeanList.get(1);
        }

        tvCountryName = view.findViewById(R.id.tv_country_name);
        tvCountryName.setText(countryBean.getNameId());

        tvMoneyUnit = view.findViewById(R.id.tv_money_unit);
        tvMoneyUnit.setText(countryBean.getUnitId());

        et = view.findViewById(R.id.et);
        et.addTextChangedListener(getTextWatcher());

        sdv = view.findViewById(R.id.sdv);
        sdv.setActualImageResource(countryBean.getIconId());
        sdv.setOnClickListener(view1 -> showDialog(view));

        return view;
    }

    private void showDialog(View view) {
        if (popupWindow == null) {
            View contentView = View.inflate(context, R.layout.dialog_select_country, null);
            popupWindow = new MyPopupWindow(context, contentView, ViewGroup.LayoutParams.WRAP_CONTENT, 1000);

            CountryAdapter countryAdapter = new CountryAdapter(countryBeanList);
            RecyclerView rv = contentView.findViewById(R.id.rv_country);

            rv.setAdapter(countryAdapter);
            rv.setLayoutManager(new GridLayoutManager(context, 3));
            countryAdapter.setOnClickListener((v, position) -> {
                countryBean = countryBeanList.get(position);
                sdv.setActualImageResource(countryBean.getIconId());
                tvCountryName.setText(countryBean.getNameId());
                tvMoneyUnit.setText(countryBean.getUnitId());
                popupWindow.dismiss();
            });
        }
        popupWindow.showAtLocation(view, Gravity.CENTER, 0, 100);
    }

    private TextWatcher getTextWatcher() {
        return new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
            }

            @Override
            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
                if (et.isFocused()) {
                    mainView.onTextChanged(charSequence, flag);
                }
            }

            @Override
            public void afterTextChanged(Editable editable) {
            }
        };
    }

}

File: app/src/main/java/com/kakacat/minitool/currencyconversion/CountryFragment.kt
package com.kakacat.minitool.currencyconversion

import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import android.widget.TextView
import androidx.annotation.IntRange
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.facebook.drawee.view.SimpleDraweeView
import com.kakacat.minitool.R
import com.kakacat.minitool.common.ui.RecycleViewListener.OnItemClick
import com.kakacat.minitool.common.ui.view.MyPopupWindow
import com.kakacat.minitool.currencyconversion.model.CountryBean

class CountryFragment(private val mainView: Contract.View, private val countryBeanList: List<CountryBean>, @IntRange(from = 1,to = 2) private val flag: Int) : Fragment() {

    lateinit var et: EditText
    lateinit var countryBean: CountryBean

    private lateinit var tvCountryName: TextView
    private lateinit var tvMoneyUnit: TextView
    private lateinit var sdv: SimpleDraweeView
    private var popupWindow: MyPopupWindow? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.country_view_layout, container, false)

        fun initView(){
            if (flag == 1) {
                countryBean = countryBeanList[0]
            } else if (flag == 2) {
                countryBean = countryBeanList[1]
            }
            tvCountryName = view.findViewById(R.id.tv_country_name)
            tvMoneyUnit = view.findViewById(R.id.tv_money_unit)
            et = view.findViewById(R.id.et)
            sdv = view.findViewById(R.id.sdv)
        }

        fun setView(){
            tvCountryName.setText(countryBean.nameId)
            tvMoneyUnit.setText(countryBean.unitId)
            et.addTextChangedListener(textWatcher)
            sdv.setActualImageResource(countryBean.iconId)
            sdv.setOnClickListener { showDialog(view) }
        }

        initView()
        setView()

        return view
    }

    private fun showDialog(view: View) {
        fun initView(){
            val contentView = View.inflate(context, R.layout.dialog_select_country, null)
            popupWindow = MyPopupWindow(context!!, contentView, ViewGroup.LayoutParams.WRAP_CONTENT, 1000)
            val countryAdapter = CountryAdapter(countryBeanList)
            val rv: RecyclerView = contentView.findViewById(R.id.rv_country)
            rv.adapter = countryAdapter
            rv.layoutManager = GridLayoutManager(context, 3)
            countryAdapter.setOnClickListener(object : OnItemClick{
                override fun onClick(v: View?, position: Int) {
                    countryBean = countryBeanList[position]
                    sdv.setActualImageResource(countryBean.iconId)
                    tvCountryName.setText(countryBean.nameId)
                    tvMoneyUnit.setText(countryBean.unitId)
                    popupWindow!!.dismiss()
                }
            })
        }

        if(popupWindow == null){
            initView()
        }
        popupWindow!!.showAtLocation(view, Gravity.CENTER, 0, 100)
    }

    private val textWatcher: TextWatcher
        get() = object : TextWatcher {
            override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
            override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {
                if (et.isFocused) {
                    mainView.onTextChanged(charSequence, flag)
                }
            }

            override fun afterTextChanged(editable: Editable) {}
        }

}